tests/integration/test.design_docs.js

```diff
@@ -96,6 +96,39 @@ adapters.forEach(function (adapter) {
       });
     });
 
+    it('Indexing event', function (done) {
+      var docs1 = [
+        doc,
+        {_id: 'dale', score: 3},
+        {_id: 'mikeal', score: 5},
+        {_id: 'max', score: 4},
+        {_id: 'nuno', score: 3},
+      ];
+      var db = new PouchDB(dbs.name);
+      // Test invalid if adapter doesnt support mapreduce
+      if (!db.query || adapter !== 'local') {
+        return done();
+      }
+
+      let indexingEvents = [];
+
+      db.on('indexing', function (result) {
```

Comment:
This seems like a brittle assumption, are we guaranteed these events will fire by the time the bulkdocs then query is complete?

Reply:
Hi @daleharvey, I work with @AlbaHerrerias at Neighbourhoodie. I'm still learning about how this code works, but my assumption going into this was that calling `db.query()` first causes Pouch to make sure the underlying index is up to date with the primary data store, so any `indexing` events would necessarily be emitted before `query()` returns a result.

If that's not the case, can you give an example of when this ordering would be violated and suggest a better place for us to wait until all the expected `indexing` events have been emitted? My only other idea was to put a counter in the `indexing` event listener and invoke the assertions and `done()` when it reaches `2`.

Reply:
hey @daleharvey , thanks for you previous efforts, Travis is works and all tests are passing. Any other concern you see with the rewrite?